lib/ansible/module_utils/powershell.ps1

```diff
@@ -67,23 +67,26 @@ Function Exit-Json($obj)
 }
 
 # Helper function to add the "msg" property and "failed" property, convert the
-# powershell object to JSON and echo it, exiting the script
+# powershell Hashtable to JSON and echo it, exiting the script
 # Example: Fail-Json $result "This is the failure message"
 Function Fail-Json($obj, $message = $null)
 {
-    # If we weren't given 2 args, and the only arg was a string, create a new
-    # psobject and use the arg as the failure message
-    If ($message -eq $null -and $obj.GetType().Name -eq "String")
-    {
+    If (-not $obj -or -not $obj.GetType) {
+        # If the first argument is undefined or not an object, make it a Hashtable
+        $obj = @{ }
+    } ElseIf ($obj.GetType().Name -eq "PSCustomObject" -or $obj.GetType().Name -eq "Hashtable") {
```

Comment:
Would be nice to use the -is test here 
so
`$obj -is [psobject] -or $obj -is [hashtable]`
instead of the `.GetType().Name -eq "name of type as a string"` syntax - in fact that ought to be another thing for ansible powershell style guide IMO.